api/api.go

```diff
@@ -28,23 +37,24 @@ type API struct {
 	staticRouter *httprouter.Router
 }
 
+// ctxValue is a helper type which makes it safe to register values in the
+// context. If we don't use a custom unexported type it's possible for others
+// to get our value.
+type ctxValue string
```

Comment:
I haven't really played around with this but it sounds like a fun exploration task.
For the moment I changed the docstring to say `easy` instead of `possible`.